### ISSUE TITLE
Add the ability to reopen closed tabs

### ISSUE THREAD

**Issue by brismuth** [thomaswilburn/Caret#451]
It would be really nice to be able to reopen recently closed tabs with Ctrl+Shift+t, the way you can in sublime and chrome. It shouldn't be too difficult to keep an array of recently opened files (node entries) that is only kept in memory and not persisted.

**Comment by thomaswilburn** [comment]
Only the Project Manager actually uses those FSNode objects. I would think for this, you could keep a stack of Tab objects that have been detached from the editor, and pop them off when someone asks to re-open a tab. We already keep two stacks in the session state, one for Ctrl-tab order and one for visible tab order. Adding a third alongside this should be a pretty easy thing to do.

**Comment by sami10007** [comment]
I think this is a problematic proposal. I would like to have TextMate-like view about getting a list of recent files. Its common keybinding is CTRL-SHIFT+T. This view is much more important for me than the feature about reopening closed tabs.

**Comment by thomaswilburn** [comment]
See also the comments on #456 - "recent files" is a difficult feature to add reliably, since we can't guarantee that we can actually re-open retained file handles. I'm wary of adding features that may fail on edge cases that are totally out of our control.

There are a lot of things like this in Caret that seem weird, or like I'm being a stick in the mud, but they're actually tied to some of the weird nature of the packaged app APIs. File retention and paths are both difficult for us, due to security precautions built into the Chrome runtime. It's not as easy as it seems.

**Comment by brismuth** [comment]
@sami10007 I'm not sure I follow. What makes it a problematic proposal? Or do you just mean for the reasons that @thomaswilburn said?

@thomaswilburn that makes sense. I think for this one it should be okay though, since we won't be persisting the tab list across reloads. Also, since we're reusing the same exact tab object, isn't it equally likely that we'd lose retention of a currently open tab?

**Comment by sami10007** [comment]
@brismuth Yes, for the reasons of Thomas but also because the keybinding CTRL-SHIFT-T could be better for view about recent-files.

**Comment by thomaswilburn** [comment]
@sami10007 I'm not particularly interested in keeping key compatibility with Textmate, to be honest with you.

@brismuth Yes, since we don't persist the list across reloads, file retention isn't an issue here at all. It's only an issue for an actual "recent files" menu that would be populated across multiple app launches.

I don't generally worry about retention IDs for tabs, because they tend to be accessed regularly and so they'll be at the top of Chrome's internal access list.

**Comment by brismuth** [comment]
I agree, Caret is intended to be a Sublime Text clone (see Readme), not a Textmate clone.

It sounds like from a technical standpoint this feature should be fairly stable then, as long as we don't have a persisted recent files menu.
